### PR TITLE
Feature/add persons search and get

### PR DESCRIPTION
**Pull request by mtarnovan**
This is in support of a drive-by change while working on https://steadymedia.atlassian.net/browse/SD-433, not directly related to the ticket: refactoring `Ypsilon.Pipedrive.Person.get_by_user_id` to use a search + a get instead of fetching all persons.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ commands:
       - restore_cache:
           name: Restore PLT cache
           key: plt-cache-v1-{{ checksum "ERLANG_VERSION" }}-{{ checksum "ELIXIR_VERSION" }}
-      - run: mix dialyzer --halt-exit-status
+      - run: mix dialyzer
```
Comment (angelikatyborska) [inline .circleci/config.yml:18]: 👍

File: lib/pipedrive/helpers/rate_limit.ex [hunk 2]
```diff
@@ -35,7 +35,7 @@ defmodule Pipedrive.Helpers.RateLimit do
   end
 
   defp execute_request(api_call, api_call_args) do
-    case apply(api_call, api_call_args) do
+    case apply(api_call, List.wrap(api_call_args)) do
```
Comment (angelikatyborska) [inline lib/pipedrive/helpers/rate_limit.ex:38]: All of the tests still pass if I remove the call to `List.wrap`. What is it for?

**Review by angelikatyborska — APPROVED** [review]
I can't believe they only offer to look for a value in all of the custom fields, and not a chosen one. But I think it won't hurt us.

File: lib/pipedrive/helpers/rate_limit.ex (hunk 2 above)
Comment (mtarnovan) [reply lib/pipedrive/helpers/rate_limit.ex:38]: It's just a cosmetic additions, it allows passing a single scalar as an argument (without wrapping it in a list), so with these both of these would work:

```elixir
sleep_and_retry(&Person.get/1, "person_id_123")
sleep_and_retry(&Person.get/1, ["person_id_123"])
```

**Pull request merged**